src/ResourceManager/Common/Commands.ResourceManager.Common/Paging/PageEnumerator.cs

```diff
@@ -17,23 +17,36 @@
 using System.Collections.Generic;
 using System.Collections;
 
-namespace Microsoft.Azure.Internal.Subscriptions.Models
+namespace Microsoft.Azure.Commands.ResourceManager.Common.Paging
 {
-    public class PageEnumerator : IEnumerator<Subscription>
+    public class PageEnumerator<T> : IEnumerator<T>
```

Comment:
Depending on what we hear from powershell core, you will need to place a constraint here, like ```where T : IPSPagingOutput```